src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadResolver.cs

```diff
@@ -0,0 +1,212 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.Net.Sdk.WorkloadManifestReader
+{
+    /// <remarks>
+    /// This very specifically exposes only the functionality needed right now by the MSBuild workload resolver
+    /// and by the template engine. More general APIs will be added later.
+    /// </remarks>
+    public class WorkloadResolver : IWorkloadResolver
+    {
+        readonly Dictionary<string, WorkloadDefinition> workloads = new Dictionary<string, WorkloadDefinition>(StringComparer.OrdinalIgnoreCase);
+        readonly Dictionary<string, WorkloadPack> packs = new Dictionary<string, WorkloadPack>(StringComparer.OrdinalIgnoreCase);
+
+        public static WorkloadResolver Create(IWorkloadManifestProvider manifestProvider)
+        {
+            var resolver = new WorkloadResolver();
+            var manifests = new List<WorkloadManifest>();
+
+            foreach (var manifestStream in manifestProvider.GetManifests())
+            {
+                using (manifestStream)
+                {
+                    var manifest = WorkloadManifestReader.ReadWorkloadManifest(manifestStream);
+                    manifests.Add(manifest);
+                }
+            }
+
+            foreach (var manifest in manifests)
+            {
+                foreach (var workload in manifest.Workloads)
+                {
+                    resolver.workloads.Add(workload.Key, workload.Value);
+                }
+                foreach (var pack in manifest.Packs)
+                {
+                    resolver.packs.Add(pack.Key, pack.Value);
+                }
+            }
+
+            return resolver;
+        }
+
+        /// <summary>
+        /// Gets the installed workload packs of a particular kind
+        /// </summary>
+        /// <remarks>
+        /// Used by MSBuild resolver to scan SDK packs for AutoImport.props files to be imported.
+        /// Used by template engine to find templatees to be added to hive.
+        /// </remarks>
+        public IEnumerable<PackInfo> GetInstalledWorkloadPacksOfKind(WorkloadPackKind kind)
```

Comment:
This API has disk IO. It would be hard to test. Maybe pass in a Func for File.Exists, and if the Func is null then use the real thing?(Poor man's DI)

Normal CLI has an in memory implementation of File and Directory. But we should not add more dependencies

Reply:
And packs are not accessible from the outside. Most of the functionality is not possible to test now

Reply:
I added IVT so the tests could still access everything.

Maybe the tests could poke a delegate into the WorkloadResolver via an internal property, which would override the use of `File.Exists` if present.

Reply:
Also make sure pack Path is constructed as

`Path.Combine(dotnetRootPath, "template-packs", pack.Id.ToLower() + "." + pack.Version.ToLower() + ".nupkg"));`

1. I verified with nuget team, `pack.Id.ToLower() + "." + pack.Version.ToLower() + ".nupkg"` is reliable to construct nupkg name from package id and package version.
2. "templates" and "template-packs" are different folders. For template packs it should be in "template-packs". "templates" is for existing non workload templates

Reply:
And normalized package id should be all lower case. Although part of the system is not consistent about it